components-core/src/main/java/com/adyen/checkout/components/core/internal/analytics/data/remote/DefaultAnalyticsSetupProvider.kt

```diff
@@ -57,8 +61,16 @@ internal class DefaultAnalyticsSetupProvider(
         is AnalyticsSource.PaymentComponent -> source.paymentMethodType
     }
 
+    private fun getLevelQueryParameter(analyticsParamsLevel: AnalyticsParamsLevel) = when (analyticsParamsLevel) {
```

Comment:
What do you think about making this a field of `AnalyticsParamLevel` enum? Then we wouldn't have to map it and just access it like `analyticsParamsLevel.queryParam`. Although it would increase coupling of this code and `AnalyticsParamLevel`, do you see a problem with that?

Reply:
I added the mapping in `DefaultAnalyticsSetupProvider`, since it is purely part of the analytics setup call, and here we define what needs to be sent as part of the initial request based on the passed level. On the other hand `AnalyticsParamsLevel` is used in Analytics manager to decide what events to send.

I prefer keeping the mapping in the `DefaultAnalyticsSetupProvider` and keeping the `AnalyticsParamsLevel` as is. It will make it easier to adjust or change the initial analytics call in the future without touching `AnalyticsParamsLevel`.

Let me know what you think.